concept/type_/attribute_type.rs

```diff
@@ -177,78 +224,81 @@ impl<'a> AttributeType<'a> {
         type_manager.set_label(snapshot, self.clone().into_owned(), label)
     }
 
-    pub fn get_supertype(
+    pub fn set_supertype(
         &self,
-        snapshot: &impl ReadableSnapshot,
+        snapshot: &mut impl WritableSnapshot,
         type_manager: &TypeManager,
-    ) -> Result<Option<AttributeType<'static>>, ConceptReadError> {
-        type_manager.get_attribute_type_supertype(snapshot, self.clone().into_owned())
+        thing_manager: &ThingManager,
+        supertype: AttributeType<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        type_manager.set_attribute_type_supertype(snapshot, thing_manager, self.clone().into_owned(), supertype)
     }
 
-    pub fn set_supertype(
+    pub fn unset_supertype(
         &self,
         snapshot: &mut impl WritableSnapshot,
         type_manager: &TypeManager,
-        supertype: AttributeType<'static>,
+        thing_manager: &ThingManager,
     ) -> Result<(), ConceptWriteError> {
-        type_manager.set_attribute_type_supertype(snapshot, self.clone().into_owned(), supertype)
+        type_manager.unset_attribute_type_supertype(snapshot, thing_manager, self.clone().into_owned())
     }
 
-    pub fn get_supertypes<'m>(
+    pub(crate) fn is_independent(
         &self,
         snapshot: &impl ReadableSnapshot,
-        type_manager: &'m TypeManager,
-    ) -> Result<MaybeOwns<'m, Vec<AttributeType<'static>>>, ConceptReadError> {
-        type_manager.get_attribute_type_supertypes(snapshot, self.clone().into_owned())
+        type_manager: &TypeManager,
+    ) -> Result<bool, ConceptReadError> {
+        type_manager.get_attribute_type_is_independent(snapshot, self.clone())
     }
 
-    pub fn get_subtypes<'m>(
+    // TODO: Might call it "get_regex", "get_range", "get_values", but "get_values" sounds like
```

Comment:
Unsure about the naming =)

Reply:
The naming is not bad at all - i would just flip them around to be `get_constraint_xxx` instead of `get_xxx_constraint` so the groupings become obvious when you're reading!

Reply:
Agree, done.